### PR TITLE
[PR1] DEV-30 MVP-2 Backend

### PR DESCRIPTION
**Pull request by jmaralo**
This PR creates the main file that orchestrates the backend for the MVP-2 and fixes some conflicts between modules

### CODE REVIEW

**Comment by jmaralo** [comment]
this PR is a certified programmer moment :)

**Comment by jmaralo** [comment]
Will separate the front from the back in future commits, as of now the working frontend will stay in cmd/MVP-2/static so that it's easier to distribute

File: src/Shared/Logger/infra/file_handler.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	"path"
+)
+
+type FileData struct {
+	handle *os.File
+	values []ValueTimestamp
+}
+
+func NewFile(parentPath string, name string) *FileData {
+	file, err := os.Create(path.Join(parentPath, fmt.Sprintf("%s.csv", name)))
```
Comment (lipezaballa) [inline src/Shared/Logger/infra/file_handler.go:16]: Why do we use .csv?

**Review by smorsua — APPROVED** [review]

File: src/Shared/Logger/infra/logger.go [hunk 2]
```diff
@@ -1,137 +1,62 @@
...
+	}()
+}
+
+func (log Logger) record() {
+	log.currDir = NewDir(path.Join(log.baseDir, time.Now().Format("2006-01-02_15-04-05")))
+loop:
+	for {
+		select {
+		case <-log.ticker.C:
+			log.currDir.Dump()
```
Comment (lipezaballa) [inline src/Shared/Logger/infra/logger.go:45]: what does "lod.currDir.Dump" do?

File: src/Shared/StreamingHandlers/streaming.go [hunk 3]
```diff
@@ -1,67 +1,63 @@
...
+			}
+		}
+	}()
+}
+
+func OrderSocketHandler(ws websocket.Conn, orderWAChannel chan OrderWebAdapter) {
+	go func() {
+		for {
+			orderWA := OrderWebAdapter{}
+			ws.ReadJSON(orderWA)
```
Comment (lipezaballa) [inline src/Shared/StreamingHandlers/streaming.go:35]: what this function does, where it waits for an order to arrive to send?

**Review by lipezaballa — APPROVED** [review]

File: src/Shared/Logger/infra/file_handler.go (hunk 1 above)
Comment (smorsua) [reply src/Shared/Logger/infra/file_handler.go:16]: We spoke to electromagnetics and agreed on that format. They can import it in matlab easily.

File: src/Shared/StreamingHandlers/streaming.go (hunk 3 above)
Comment (smorsua) [reply src/Shared/StreamingHandlers/streaming.go:35]: ReadJSON waits for data to come from the websocket (ws) and when it gets it, it puts it inside orderWA

File: src/Shared/Logger/infra/logger.go (hunk 2 above)
Comment (smorsua) [reply src/Shared/Logger/infra/logger.go:45]: Each file has a corresponding array that fills up with entries, and every once in a while the content from these arrays is copied into the corresponding file. Dump puts the contents of each array in the corresponding file.

**Pull request merged**